test/suite/test_prepare_hs03.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-2020 MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+from helper import copy_wiredtiger_home
+import unittest, wiredtiger, wttest
+from wtdataset import SimpleDataSet
+import os, shutil
+
+def timestamp_str(t):
+    return '%x' % t
+
+# test_prepare_hs03.py
+# test to ensure salvage, verify & simulating crash are working for prepared transactions.
+class test_prepare_hs03(wttest.WiredTigerTestCase):
+    # Force a small cache.
+    conn_config = 'cache_size=50MB'
+
+    # Create a small table.
+    uri = "table:test_prepare_hs03"
+
+    def corrupt_file(self):
+        filename="test_prepare_hs03.wt"
+        self.assertEquals(os.path.exists(filename), True)
+
+        with open(filename, 'r+') as log:
+            log.seek(1024)
+            log.write('Bad!' * 1024)
+
+    def corrupt_salvage_verify(self):
+        self.corrupt_file()
+        self.session.salvage(self.uri, "force")
+        self.session.verify(self.uri, None)
+
+    def prepare_updates(self, ds, nrows, nsessions, nkeys):
+        # Insert some records with commit timestamp, corrupt file and call salvage, verify before checkpoint.
+
+        # Commit some updates to get eviction and history store fired up
+        bigvalue1 = b"bbbbb" * 2
+        cursor = self.session.open_cursor(self.uri)
+        for i in range(1, nkeys):
+            self.session.begin_transaction('isolation=snapshot')
+            tkey = 'C' + ds.key(nrows + i)
```

Comment:
The key here has a `'C'` prefix, which means it won't match the original keys.  So these changes are not updates, and they won't create "old" versions of values, and thus won't generate history store activity.  Is there some other reasoning for using a new key?

Reply:
@ddanderson : Removed the prefix and changed the code to update the same key with different values for history store activity.